tnrg 4.28 clang 19 compiler error: "constexpr variable 'n' must be initialized by a constant expression"

```
/usr/WS2/lidoext/libs/lido/toss_4_x86_64_ib/upstreams/2026_03_12_15_35_42/linux-rhel8-ivybridge/llvm-19.1.3/trng-4.28-lw5si5rngmo4p5cothonvuxfhty7diik/include/trng/special_functions.hpp:365:31: error: constexpr variable 'n' must be initialized by a constant expression
  365 |         constexpr std::size_t n{coeffs.max_index() - 1};
      |                               ^~~~~~~~~~~~~~~~~~~~~~~~~
/usr/WS2/lidoext/libs/lido/toss_4_x86_64_ib/upstreams/2026_03_12_15_35_42/linux-rhel8-ivybridge/llvm-19.1.3/trng-4.28-lw5si5rngmo4p5cothonvuxfhty7diik/include/trng/special_functions.hpp:395:29: note: in instantiation of function template specialization 'trng::math::detail::GammaPQ_asympt_R<float>' requested here
  395 |         const T correction{-GammaPQ_asympt_R(a, eta, eta_squared_half)};
      |                             ^
/usr/WS2/lidoext/libs/lido/toss_4_x86_64_ib/upstreams/2026_03_12_15_35_42/linux-rhel8-ivybridge/llvm-19.1.3/trng-4.28-lw5si5rngmo4p5cothonvuxfhty7diik/include/trng/special_functions.hpp:427:20: note: in instantiation of function template specialization 'trng::math::detail::GammaP_asympt<float>' requested here
  427 |             return GammaP_asympt(a, x);
      |                    ^
/usr/WS2/lidoext/libs/lido/toss_4_x86_64_ib/upstreams/2026_03_12_15_35_42/linux-rhel8-ivybridge/llvm-19.1.3/trng-4.28-lw5si5rngmo4p5cothonvuxfhty7diik/include/trng/special_functions.hpp:466:22: note: in instantiation of function template specialization 'trng::math::detail::GammaP<float, true>' requested here
  466 |       return detail::GammaP<float, true>(a, x);
      |                      ^
/usr/WS2/lidoext/libs/lido/toss_4_x86_64_ib/upstreams/2026_03_12_15_35_42/linux-rhel8-ivybridge/llvm-19.1.3/trng-4.28-lw5si5rngmo4p5cothonvuxfhty7diik/include/trng/special_functions.hpp:347:60: note: indexing of array without known bound is not allowed in a constant expression
  347 |         static constexpr T get(std::size_t index) { return d[index]; }
      |                                                            ^
/usr/WS2/lidoext/libs/lido/toss_4_x86_64_ib/upstreams/2026_03_12_15_35_42/linux-rhel8-ivybridge/llvm-19.1.3/trng-4.28-lw5si5rngmo4p5cothonvuxfhty7diik/include/trng/special_functions.hpp:351:35: note: in call to 'get(49)'
  351 |                            : (abs(get(index - 1)) > numeric_limits<T>::epsilon()
      |                                   ^~~~~~~~~~~~~~
/usr/WS2/lidoext/libs/lido/toss_4_x86_64_ib/upstreams/2026_03_12_15_35_42/linux-rhel8-ivybridge/llvm-19.1.3/trng-4.28-lw5si5rngmo4p5cothonvuxfhty7diik/include/trng/special_functions.hpp:365:33: note: in call to 'max_index(50)'
  365 |         constexpr std::size_t n{coeffs.max_index() - 1};
      |                                 ^~~~~~~~~~~~~~~~~~
```

one simple solution is to not make this variable constexpr, but maybe there is another way.

--- a/repos/spack_repo/builtin/packages/trng/package.py
+++ b/repos/spack_repo/builtin/packages/trng/package.py
@@ -15,7 +15,7 @@ class Trng(CMakePackage):
 
     maintainers("chapman39")
 
-    version("4.28", commit="691817a4a0331421776e7f00efa9fbe81554da86", submodules=True)
+    version("4.28.1", commit="cc1b170b050b541ac481415b175394670c2a4e85", submodules=True)
     version("4.24", commit="a22a32b9a285d1293b74b17b34f81af5dcec6311")
     version("4.23.1", commit="610f7836610e9e01788f8461fbe83b67544ded7c")
     version("4.23", commit="22a30c493eff4cb547185fa48275683751df18e0")

